sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_client.py

```diff
@@ -326,24 +337,26 @@ def purge_deleted_key(self, name, **kwargs):
             .. code-block:: python
 
                 # if the vault has soft-delete enabled, purge permanently deletes a deleted key
-                # (with soft-delete disabled, delete_key is permanent)
+                # (with soft-delete disabled, begin_delete_key is permanent)
                 key_client.purge_deleted_key("key-name")
 
         """
         self._client.purge_deleted_key(vault_base_url=self.vault_endpoint, key_name=name, **kwargs)
 
     @distributed_trace
-    def recover_deleted_key(self, name, **kwargs):
+    def begin_recover_deleted_key(self, name, **kwargs):
         # type: (str, **Any) -> KeyVaultKey
-        """Recover a deleted key to its latest version. This is only possible in vaults with soft-delete enabled. If a
-        vault does not have soft-delete enabled, :func:`delete_key` is permanent, and this method will return an error.
-        Attempting to recover an non-deleted key will also return an error.
+        """Recover a deleted key to its latest version. This is only possible in vaults with soft-delete enabled.
+        If a vault does not have soft-delete enabled, :func:`begin_delete_key` is permanent, and this method will
+        return an error. Attempting to recover an non-deleted key will also return an error.
 
         Requires the keys/recover permission.
 
         :param str name: The name of the deleted key
-        :returns: The recovered key
-        :rtype: ~azure.keyvault.keys.KeyVaultKey
+        :returns: An poller for the recover key operation. Calling result() on the poller will return the recovered
```

Comment:
It might be clearer here to describe `wait()` as waiting until the recovered key is *ready to use*.

Reply:
added that, thanks!